pkg/eraser/eraser_test.go

```diff
@@ -0,0 +1,382 @@
+	s := c.images
+	s = append(s[:index], s[index+1:]...)
+	c.images = s
+}
+
+func (c *testClient) removeImage(ctx context.Context, image string) (err error) {
+	if image == "" {
+		return ErrImageEmpty
+	}
+
+	for index, value := range c.images {
+		for _, repotag := range value.RepoTags {
+			if value.Id == image || repotag == image {
+				c.removeImageFromSlice(index)
+				return nil
+			}
+		}
+
+		for _, repodigest := range value.RepoDigests {
+			if repodigest == image {
+				c.removeImageFromSlice(index)
+				return nil
+			}
+		}
+	}
+	return ErrImageNotRemoved
+}
+
+func testEqImages(a, b []*pb.Image) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	auxMap := make(map[string]bool, len(a))
+	for _, i := range a {
+		auxMap[i.Id] = true
+	}
+	for _, j := range b {
+		if auxMap[j.Id] == false {
+			return false
+		}
+	}
+	return true
+}
+
+func testEqContainers(a, b []*pb.Container) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	auxMap := make(map[string]bool, len(a))
+	for _, i := range a {
+		auxMap[i.Id] = true
+	}
+	for _, j := range b {
+		if auxMap[j.Id] == false {
+			return false
+		}
+	}
+	return true
+}
+
+func TestListImages(t *testing.T) {
+	var testCases = []struct {
+		imagesInput  testClient
+		imagesOutput []*pb.Image
+		err          error
+	}{
+		{
+			imagesInput: testClient{
+				containers: []*pb.Container{},
+				images:     []*pb.Image{&image1, &image2, &image3, &image4, &image5},
+			},
+			imagesOutput: []*pb.Image{&image1, &image2, &image3, &image4, &image5},
+			err:          nil,
+		},
+		{
+			imagesInput: testClient{
+				containers: []*pb.Container{},
+				images:     []*pb.Image{},
+			},
+			imagesOutput: []*pb.Image{},
+			err:          nil,
+		},
+	}
+
+	backgroundContext, cancel := context.WithTimeout(context.Background(), timeoutTest)
+	defer cancel()
+
+	for _, tc := range testCases {
+		l, e := tc.imagesInput.listImages(backgroundContext)
+		if testEqImages(l, tc.imagesOutput) == false || !errors.Is(e, tc.err) {
+			t.Errorf("Test fails")
+		}
+	}
+
+}
+
+func TestListContainers(t *testing.T) {
+	var testCases = []struct {
+		containersInput testClient
+		containerOutput []*pb.Container
+		err             error
+	}{
+		{
+			containersInput: testClient{
+				containers: []*pb.Container{&container1, &container2},
+				images:     []*pb.Image{},
+			},
+			containerOutput: []*pb.Container{&container1, &container2},
+			err:             nil,
+		},
+		{
+			containersInput: testClient{
+				containers: []*pb.Container{},
+				images:     []*pb.Image{},
+			},
+			containerOutput: []*pb.Container{},
+			err:             nil,
+		},
+	}
+
+	backgroundContext, cancel := context.WithTimeout(context.Background(), timeoutTest)
+	defer cancel()
+
+	for _, tc := range testCases {
+		l, e := tc.containersInput.listContainers(backgroundContext)
+		if testEqContainers(l, tc.containerOutput) == false || !errors.Is(e, tc.err) {
+			t.Errorf("Test fails")
+		}
+	}
+
+}
+
+func TestRemoveImage(t *testing.T) {
```

Comment:
can we add a test for running images too? try to delete but it shouldn't delete

Reply:
Yes! I added it. Before I was not checking if the image was running before removing so I added that functionality and changed the unit tests.